public/app/features/alerting/unified/api/alertSilencesApi.ts

```diff
@@ -26,8 +26,14 @@ export const alertSilencesApi = alertingApi.injectEndpoints({
           accesscontrol: accessControl,
         },
       }),
-      providesTags: (result) =>
-        result ? result.map(({ id }) => ({ type: 'AlertmanagerSilences', id })) : ['AlertmanagerSilences'],
+      providesTags: (result) => {
+        return result
+          ? [
+              ...result.map(({ id }) => ({ type: 'AlertmanagerSilences' as const, id })),
+              { type: 'AlertmanagerSilences' },
```

Comment:
The gist here is adding the generic `{ type: 'AlertmanagerSilences' }` also when we do have a result